### ISSUE TITLE
Code 20041 MI id is not associated with a phone number, but I can't add any

### ISSUE THREAD

**Issue by mwauer** [offici5l/MiUnlockTool#40]
Probably not a bug of the tool, but I'm stuck and don't know how to proceed.
I got this code when trying to unlock. Unfortunately I can't add a recovery phone to my MI account. The options suggested in https://account.xiaomi.com/helpcenter/faq/en_US/03.account-safety/04.security-phone-and-email/faq-3 aren't there. I even tried creating a new MI account on another phone so it links to it, but it didn't work.
Does anybody have an idea on how to do this? Thanks in advance!

**Comment by offici5l** [comment]
Most companies currently interact with whatsapp regarding OTP, so if your phone number is linked to a whatsapp account, the confirmation code is likely to be sent to the whatsapp account rather than the phone itself. 

If you encounter the same issue, try using a temporary number. you can find many websites on google that offer this service for free. try different numbers until you get a valid one.

**Comment by Ashvith10** [comment]
@offici5l there's no way to add any number. That's the actual issue.

**Comment by Ashvith10** [comment]
It looks like there is: https://offici5l.github.io/articles/mi-error-codes.html. I'll check this tomorrow once the code cool-down for Mi account is over.

**Comment by Hahatriceps** [comment]
Hey Guys, im a little late here, but I was able to find a workaround :

Setup: Redmi Note 9 Pro (curtana), region IN, Mi account = email only. Was on MIUI 12.5.5 Global Stable → updated to 14.0.3 Global Stable (the update that surfaced the fix).
Start

Wanted bootloader unlock (custom ROM / Linux), used official Mi Unlock + [offici5l/MiUnlockTool].

Stuck

Official Mi Unlock (v6.5.224.28) → blank "Account Authentication" page (ancient embedded browser).
offici5l/MiUnlockTool (terminal login, SMS/email) → worked, detected curtana.
Final step → 20041: MI ID not associated with a phone number.
Email-only account, no option to add a phone anywhere: id.mi.com, on-device settings, even a new account.
Different SIM/number useless — no field to enter one. Account delete blocked by Find Device.

Solution

Phone option appeared as a system prompt, not in settings.
Updated 12.5.5 → 14.0.3 → enabled Mi services (Find Device) → opened stock SMS app → got "add recovery phone" prompt → registered + verified number.
Number now under recovery phone → error went 20041 → 20036 (164h wait), i.e. requirement cleared.
⚠ Don't re-add account / re-bind during the wait — resets the 168h timer.
